### PR TITLE
fix billboard rendering

### PR DESCRIPTION
**Pull request by star-e**
Re: #16504

### Changelog

* add technique to the billboard.ts

Default rendering technique is 0, and will do additive blending.
Now user can choose technique in billboard component. 
For example, technique 1 will do alpha blend.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

File: cocos/particle/billboard.ts [hunk 1]
```diff
@@ -113,6 +113,54 @@ export class Billboard extends Component {
...
+
+    public set technique (val: number) {
+        // clamp technique index
+        val = Math.floor(val);
+        if (val < 0) {
+            val = 0;
+        }
+        const techs = this._material?.effectAsset?.techniques;
+        if (techs && val >= techs.length) {
+            val = techs.length - 1;
```
Comment (minggo) [inline cocos/particle/billboard.ts:145]: Is it a problem if techs.length is 0?

File: cocos/particle/billboard.ts [hunk 2]
```diff
@@ -113,6 +113,54 @@ export class Billboard extends Component {
...
+            this._material = null;
+            this._mesh.destroy();
+            this._mesh = null;
+            // recreate model
+            this.createModel();
+            // set properties
+            this.width = this._width;
+            this.height = this._height;
+            this.rotation = this.rotation;
+            this.texture = this.texture;
```
Comment (minggo) [inline cocos/particle/billboard.ts]: It seems strange to set properties like this.

File: cocos/particle/billboard.ts [hunk 3]
```diff
@@ -136,6 +184,7 @@ export class Billboard extends Component {
         this.height = this._height;
         this.rotation = this.rotation;
         this.texture = this.texture;
+        this.technique = this.technique;
```
Comment (minggo) [inline cocos/particle/billboard.ts]: It is strange too.

File: cocos/particle/billboard.ts (hunk 1 above)
Comment (star-e) [reply cocos/particle/billboard.ts:145]: Yeah, have never thought of. Theoretically it is possible.
I will change the code.

File: cocos/particle/billboard.ts (hunk 2 above)
Comment (star-e) [reply cocos/particle/billboard.ts]: Refactored these assignments

**Review by minggo — APPROVED** [review]

**Pull request merged**